src/streamsync/core.py

```diff
@@ -222,19 +222,40 @@ def __setitem__(self, key, raw_value) -> None:
         self.state[key] = value
         self.apply(key)
 
+    def __delitem__(self, key: str) -> None:
```

Comment:
This works for explicitly deleted items. But are we covering the case when someone regenerates a new `StateProxy` like Fabien showed in his code (when he used a dictionary comprehension)? 

For example:

`state["animals"] = { "cat": 1, "dog": 2 }`

and then in the second run:

`state["animals"] = { "dog": 3 }`

Are we somehow detecting this and notifying the front end of the deletion?

I don't think it is, please let me know. Unfortunately I think a more complex and less efficient approach will be needed.

Reply:
Well noticed. I'll verify it and post an update.

Reply:
I've tested this case, and it seems that currently mutations for nested dictionaries are processed such that they're "nullified" each time and then reconstructed with "new" content. I'm not really sure that this behavior is efficient, but it allows to account for the case you've described. So, yeah, non-explicit removal technically work too. 

Though, an interesting edge case that I've noticed to not work here is when we try to "non-explicitly remove" the whole state at once, i.e. 
```python
def very_strange_handler(state):
    state = {}
```
This simply does not produce any mutations at all, seemingly due to not invoking __setitem__ on neither `StreamsyncState` nor `StateProxy`.

Reply:
Thanks for looking into this in depth; I appreciate it. It's a complex mechanism and it's easy to miss the edge cases. Let me know whenever you have an update, feel free to take your time. Also if you want to look at it together I'm happy to.